crawl-ref/source/cio.cc

```diff
@@ -77,26 +77,31 @@ class unwind_cursor
 int unmangle_direction_keys(int keyin, KeymapContext keymap,
                             bool fake_ctrl, bool fake_shift)
 {
+    const bool use_modifier_prefix_keys = Options.use_modifier_prefix_keys;
```

Comment:
What is this local for? You could just use `Options.use_modifier_prefix_keys` directly in the `if` condition.

Alternatively, I think it would be even better to combine the `fake_ctrl` and `fake_shift` arguments into a single `allow_fake_modifiers` argument (since they are never actually passed differently). In that case, you could simplify the decision tree:

```
if (allow_fake_modifiers && Options.use_modifier_prefix_keys)
{
    if (keyin == '*')
    {
        ...
    }
    ...
}
```